freifunkMUC/community-packages
# revalidate Gateway selection

- Test if node is still on the best GW
- Move Duplicate code in functions.sh
- add new Function in functions.sh (use_api_best_gw, init_vars, check_mesh_vpn, check_wireguardkey, check_NTP, setup_connection)
- use_api_best_gw to get the best gateway from the broker(atm v2 Endpoint, needs a new Endpoint on the broker) including a bool if the nodes that checks should switch (generally if GW are not LoadBalanced)
- check-bestgateway.sh: Script to check and do something if broker response with we should switch
- check current peer public key againts public key of best gw

the check if are we on the best gateway would produce some background traffic every x mintues so it would be good to disable the check on the client? (Web gui?)

More Ideas how we can do this:
Send the Peer Publicy Key/Name or something to the broker and only send clients the bool(should switch) if are affected to switch gw (Client should check nevertheless if the best gw public key is a different public key)
